targets/TARGET_STM/lp_ticker.c

```diff
@@ -272,9 +298,34 @@ void lp_ticker_set_interrupt(timestamp_t timestamp)
     if (lp_cmpok == false) {
         /* if this is not safe to write, then delay the programing to the
          * time when CMPOK interrupt will trigger */
+        
+        /* If this target timestamp is close to the roll over of the ticker counter
+         * and current tick is also close to the roll over, then we are in danger zone.*/
+        if(((0xFFFF - LP_TIMER_SAFE_GUARD < timestamp) || (timestamp < LP_TIMER_SAFE_GUARD))
+                && ((0xFFFA < last_read_counter) || (last_read_counter < LP_TIMER_SAFE_GUARD)))
```

Comment:
Why is the condition `(last_read_counter < LP_TIMER_SAFE_GUARD)` also needed ? 
If this is the case, then the rollover already happened, right ?
I don't think it would cause an issue though ...

Reply:
You are right. It could do some mess if you want to set timestamp to 0xFFFF, or something close to, and current tick is at the beginning of the counter.